### PR TITLE
Redis Sentinel support

### PR DESCRIPTION
**Pull request by troll-os**
### Proposed changes

* Added options to connect to a Redis Sentinel cluster 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

I didn't actually write tests as no functional behaviors were altered in this PR. Though the technical aspects of the solution in Sentinel mode were evaluated in our dev environment, feel free to reach out so we can elaborate tests if deemed necessary, the instance is still up.

Relevant new env vars are in a yet to push commit on the docs repo, waiting for this PR to be merged first.

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/database/redis.ts [hunk 1]
```diff
@@ -90,14 +90,26 @@ const clusterOptions = (): ClusterOptions => ({
...
+  const redisMode: string = conf.get('redis:mode');
+
+  if (redisMode === 'cluster') {
     const clusterNodes = generateClusterNodes(conf.get('redis:hostnames') ?? []);
     client = new Redis.Cluster(clusterNodes, clusterOptions());
+  } else if (redisMode === 'sentinel') {
+    const sentinelConfiguration: RedisOptions = {
+      sentinels: [
+        { host: conf.get('redis:sentinel_leader_host'), port: conf.get('redis:sentinel_leader_port') },
+        { host: conf.get('redis:sentinel_additional_host'), port: conf.get('redis:sentinel_additional_port') },
```
Comment (richard-julien) [inline opencti-platform/opencti-graphql/src/database/redis.ts]: Not possible to have a list instead of limiting to 2 entries?

Comment (troll-os) [reply opencti-platform/opencti-graphql/src/database/redis.ts]: Yep could be, don't know how that didn't hit me before pushing... Can even reuse the generateClusterNodes() func.

Has some impact on internal platform definitions, will update the PR when everything is aligned

**Comment by troll-os** [comment]
Small code refacto done + infra tests done on our dev envs, the platform didn't suffer any availability impact

**Comment by richard-julien** [comment]
Untested but code looks good! 
Really love to see some OpenCTI platform team PR. Thanks!

**Review by richard-julien — APPROVED** [review]

**Pull request merged**